packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIcon.tsx

```diff
@@ -8,44 +8,108 @@ import { useTheme } from '~components/BladeProvider';
 import Svg, { Path } from '~components/Icons/_Svg';
 import { getIn } from '~utils';
 
-const CheckedIcon = ({ color }: { color: string }) => {
+const CheckedIconSmall = ({ color }: { color: string }) => {
   return (
-    <Svg width="16px" height="16px" viewBox="0 0 16 16" fill="none">
+    <Svg width="8" height="8" viewBox="0 0 8 8" fill="none">
       <Path
         fillRule="evenodd"
         clipRule="evenodd"
-        d="M12.3536 4.64645C12.5488 4.84171 12.5488 5.15829 12.3536 5.35355L6.85355 10.8536C6.65829 11.0488 6.34171 11.0488 6.14645 10.8536L3.64645 8.35355C3.45118 8.15829 3.45118 7.84171 3.64645 7.64645C3.84171 7.45118 4.15829 7.45118 4.35355 7.64645L6.5 9.79289L11.6464 4.64645C11.8417 4.45118 12.1583 4.45118 12.3536 4.64645Z"
+        d="M6.90237 1.76413C7.03254 1.89431 7.03254 2.10536 6.90237 2.23554L3.2357 5.90221C3.10553 6.03238 2.89447 6.03238 2.7643 5.90221L1.09763 4.23554C0.967456 4.10536 0.967456 3.89431 1.09763 3.76414C1.22781 3.63396 1.43886 3.63396 1.56904 3.76414L3 5.1951L6.43096 1.76413C6.56114 1.63396 6.77219 1.63396 6.90237 1.76413Z"
         fill={color}
+        stroke="white"
+        strokeWidth="0.5"
+        strokeLinecap="round"
+        strokeLinejoin="round"
       />
     </Svg>
   );
 };
 
-const IndeterminateIcon = ({ color }: { color: string }) => {
+const CheckedIconMedium = ({ color }: { color: string }) => {
```

Comment:
Why do we need different icons for different sizes? Won't the same svg with different height / width work 🤔

Reply:
Yeah i could, in fact I did that earlier. 
But with the width/height we also need to change the definition of the path, `<Path d="" />`. 
So i just took the simplified approach.

If this looks good, I'm fine with it: 

```ts
const IndeterminateIcon = ({ color, size }: { color: string; size: 'small' | 'medium' }) => {
  const path = {
    medium:
      'M2 6C2 5.72386 2.22386 5.5 2.5 5.5H9.5C9.77614 5.5 10 5.72386 10 6C10 6.27614 9.77614 6.5 9.5 6.5H2.5C2.22386 6.5 2 6.27614 2 6Z',
    small:
      'M1.3335 3.99984C1.3335 3.81574 1.48273 3.6665 1.66683 3.6665H6.3335C6.51759 3.6665 6.66683 3.81574 6.66683 3.99984C6.66683 4.18393 6.51759 4.33317 6.3335 4.33317H1.66683C1.48273 4.33317 1.3335 4.18393 1.3335 3.99984Z',
  };
  const dims = {
    small: {
      width: 8,
      height: 8,
      viewBox: '0 0 8 8',
    },
    medium: {
      width: 12,
      height: 12,
      viewBox: '0 0 12 12',
    },
  };

  const width = `${dims[size].width}px`;
  const height = `${dims[size].height}px`;
  const viewBox = dims[size].viewBox;
  const d = path[size];

```

Reply:
I see, I thought the same path `d` could work for all sizes but I guess not? In that case both the current implementation and the one you shared above seem fine to me.

Reply:
Yes. 

but considering that there will be another variant "large" later on, which approach do you think would be better? 🤔

Reply:
I guess we would end up anyway bundling all these icons so the one you shared here would be more preferred imo.

Reply:
Done.

Reply:
I think the d changes if you try to copy individual svg from figma for each size

Can we just use the same `d` and only change width & height. This is how svg solves such use-cases anyway. Viewbox will remain same but width & height will change. @anuraghazra can you try this once? It will simplify this logic a lot especially because we have large as well now

Reply:
This is exactly what I meant in the OP, using the same `d` 😆

Reply:
Yeah so i was also changing the "viewbox" to "0 0 8 8" so thats why needed to change the "d" also